apis/test/example/v1alpha1/fiz_types.go

```diff
@@ -103,3 +103,8 @@ func (f *ClusterFiz) GetTypeMeta() *metav1.TypeMeta {
 func (f *ClusterFiz) GetStatus() *duckv1.Status {
 	return &f.Status.Status
 }
+
+// GetHappyConditionType retrieves the happy condition of this resource. Implements the KRShaped interface.
+func (f *ClusterFiz) GetHappyConditionType() apis.ConditionType {
+	return apis.ConditionReady
```

Comment:
ooh, it is set in the api. ok I got it a bit better

Reply:
Yea downstream will set like this. After this, I'll keep incorporating here: https://github.com/knative/serving/pull/7847

Annoyingly I'll have to ensure I have the right type per every resource (which as a newbie I don't know off the top of my head)